### PR TITLE
Added GameService responsible for getting all available games, creating a new game, joining a player to a game, getting status of a game and playing a game

### PR DESCRIPTION
**Pull request by martin-nikolov**

### CODE REVIEW

File: src/TicTacToe.Services/UserService.cs [hunk 1]
```diff
@@ -50,8 +49,8 @@ public UserInfoOutput Register(UserRegistrationInput input)
 
             context.Users.Add(user);
             context.SaveChanges();
-
-            return user.ToUserInfo();
+            var result = user.ToUserInfo();
+            return result;
         }
```
Comment (martin-nikolov) [inline src/TicTacToe.Services/UserService.cs]: Just `return user.ToUserInfo()`.

File: src/TicTacToe.Services/GameResultValidator.cs [hunk 2]
```diff
@@ -69,6 +69,10 @@ private GameResult GetWinner(char playerChar)
             {
                 return GameResult.WonByO;
             }
+            else if (playerChar == '-')
+            {
+                return GameResult.NotFinished;
+            }
             else
```
Comment (martin-nikolov) [inline src/TicTacToe.Services/GameResultValidator.cs:86]: Add unit test(s) for the case with `GameResult.NotFinished`.

File: src/TicTacToe.Data/Extensions/DbContextExtensions.cs [hunk 3]
```diff
@@ -23,38 +22,6 @@ public static bool AllMigrationsApplied(this DbContext context)
 
         public static void EnsureSeeded(this TicTacToeDbContext context)
         {
```
Comment (martin-nikolov) [inline src/TicTacToe.Data/Extensions/DbContextExtensions.cs]: Delete the empty `EnsureSeeded()` method from the code.

File: src/TicTacToe.ConsoleApp/GameEngine.cs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                }
+
+                Console.WriteLine("|");
+            }
+
+            Console.WriteLine(new string('-', 9));
+        }
+
+        private (int Row, int Col) ChoosePosition(int number)
+        {
```
Comment (martin-nikolov) [inline src/TicTacToe.ConsoleApp/GameEngine.cs]: Rename the method with more appropriate name, e. g. `GetChosenPosition()` or just `GetPosition()`.

File: src/TicTacToe.ConsoleApp/GameEngine.cs [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+                case 5: return (1, 1);
+                case 6: return (1, 2);
+
+                case 7: return (0, 0);
+                case 8: return (0, 1);
+                case 9: return (0, 2);
+            }
+
+            throw new ValidationException("Invalid input.");
+        }
```
Comment (martin-nikolov) [inline src/TicTacToe.ConsoleApp/GameEngine.cs]: Extend the message with more details, e. g. `Invalid input. Please provide a valid position (row and column).`

File: src/TicTacToe.ConsoleApp/GameEngine.cs [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+                        Console.WriteLine("Enter valid number");
+                    }
+
+                    return gameService.Join(availableGames[chosenGame - 1].Id, userId).Id;
+                }
+            }
+        }
+
+        private void PrintBoard(string board)
+        {
```
Comment (martin-nikolov) [inline src/TicTacToe.ConsoleApp/GameEngine.cs:71]: Optimize the method `PrintBoard()` to use `StringBuilder`.

File: src/TicTacToe.ConsoleApp/GameEngine.cs [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+            {
+                using (var context = new TicTacToeDbContextFactory().CreateDbContext())
+                {
+                    var gameService = new GameService(context, new GameResultValidator());
+
+                    var game = gameService.Status(gameId, userId);
+                    this.PrintBoard(game.Board);
+
+                    if (game.State == GameState.WaitingForASecondPlayer)
+                    {
```
Comment (martin-nikolov) [inline src/TicTacToe.ConsoleApp/GameEngine.cs]: Separate the __if-else__ checks into a new method to improve the method readability and to minimize the complexity. Name the method, for example, `PrintState(game, userId, gameService)` and call it under the `this.PrintBoard(game.Board);`

File: src/TicTacToe.ConsoleApp/GameEngine.cs [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+                else
+                {
+                    Console.WriteLine("Choose game:");
+
+                    for (int i = 0; i < availableGames.Count; i++)
+                    {
+                        Console.WriteLine($"{i+1}. {availableGames[i].Name}");
+                    }
+
+                    int chosenGame;
```
Comment (martin-nikolov) [inline src/TicTacToe.ConsoleApp/GameEngine.cs]: Rename `chosenGame` to `chosenGameId` ?

File: src/TicTacToe.Services/GameService.cs [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+            }
+
+            boardArray[position] = playerChar;
+            string boardInsertedPosition = string.Join(string.Empty, boardArray);
+            var gameResult = gameValidator.GetGameResult(boardInsertedPosition);
+
+            game.Board = boardInsertedPosition;
+
+            if (gameResult == GameResult.NotFinished)
+            {
```
Comment (martin-nikolov) [inline src/TicTacToe.Services/GameService.cs]: Separate the if-else checks into a new method to improve the method readability and to minimize the complexity.

File: src/TicTacToe.ConsoleApp/GameEngine.cs (hunk 8 above)
Comment (davidtodorov) [reply src/TicTacToe.ConsoleApp/GameEngine.cs]: done

File: src/TicTacToe.ConsoleApp/GameEngine.cs (hunk 6 above)
Comment (davidtodorov) [reply src/TicTacToe.ConsoleApp/GameEngine.cs:71]: done

File: src/TicTacToe.ConsoleApp/GameEngine.cs (hunk 4 above)
Comment (davidtodorov) [reply src/TicTacToe.ConsoleApp/GameEngine.cs]: done

File: src/TicTacToe.ConsoleApp/GameEngine.cs (hunk 7 above)
Comment (davidtodorov) [reply src/TicTacToe.ConsoleApp/GameEngine.cs]: done

**Pull request closed without merging**